library/base/src/main/kotlin/org/cru/godtools/base/Settings.kt

```diff
@@ -157,6 +165,32 @@ class Settings internal constructor(private val context: Context, coroutineScope
     }
     // endregion Dashboard Settings
 
+    // region optInNotification
+    fun getLastPromptedOptInNotification(): LocalDate {
+        val lastPrompted = LocalDate.ofEpochDay(prefs.getLong(LAST_PROMPTED_OPT_IN_NOTIFICATION, 0))
+
+        return lastPrompted
+    }
+
+    fun getOptInNotificationPromptCount(): Int {
+        val promptCount = prefs.getInt(OPT_IN_NOTIFICATION_PROMPT_COUNT, 0)
+
+        return promptCount
+    }
```

Comment:
so a fun thing for simple functions like this in Kotlin is that it can be reduced to an expression function like this:

```suggestion
    fun getOptInNotificationPromptCount() = prefs.getInt(OPT_IN_NOTIFICATION_PROMPT_COUNT, 0)
```

Reply:
`getLastPromptedOptInNotification()` can be simplified in a similar manner

Reply:
Interesting, and no performance loss without the explicit type definition?

Reply:
nope, the compiler infers the type from the expression. you can also still specify the type if you need to.

```kotlin
fun getOptInNotificationPromptCount(): Int = prefs.getInt(OPT_IN_NOTIFICATION_PROMPT_COUNT, 0)
```

Reply:
https://kotlinlang.org/docs/functions.html#single-expression-functions